core/debugger/remote_debugger.cpp

```diff
@@ -408,7 +408,6 @@ struct RemoteDebugger::PerformanceProfiler {
 			Variant monitor_value = performance->call("get_custom_monitor", custom_monitor_names[i]);
 			if (!monitor_value.is_num()) {
 				ERR_PRINT("Value of custom monitor '" + String(custom_monitor_names[i]) + "' is not a number");
-				arr[i + max] = Variant();
```

Comment:
This change preserves what was the current behavior, but I'm not sure that it's correct.

The intent of this code's author was that in the error case, `arr[i + max]` should be `Variant()`, not a non-numeral value. So this is probably what needs to be changed (with an `else` instead of removing the assignment).

CC @simpuid who wrote that code, and @Faless who reviewed.

Reply:
Yeah, that seems to be the case. It slipped my review. Could also be `arr[i + max] = 0` instead of `Variant()`.

Reply:
[This](https://github.com/godotengine/godot/blob/5707c5d0ff67f3fa3a20ff7b30bdcd1f176b0f4b/editor/debugger/script_editor_debugger.cpp#L430) is the part where we are getting the `float` back from `Variant`
@Faless Should we add an extra check there too?

Reply:
Sorry, I missed this comment, yes, the cast should be automatic and not fail, but better be safe, so I would add the check (setting the value to 0 is not `is_num`).